From 4235903b04b681c0a12e7766695aa26f5dd34c4e Mon Sep 17 00:00:00 2001
From: Darik Horn <dajhorn@vanadac.com>
Date: Fri, 3 Feb 2012 14:49:49 -0600
Subject: [PATCH 6/7] Improve the --with-spl error for DKMS systems.

If the SPL module is unavailable, then the operator gets an error
message that does not apply to installations that are mananged by
DKMS. Change the error message to fit apt-get systems.

A better solution would be to implement a dependency model in DKMS,
or to bundle SPL into the ZFS package.
---
 config/kernel.m4 |    7 ++++---
 1 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/config/kernel.m4 b/config/kernel.m4
index a6d544b..2cb2e8c 100644
--- a/config/kernel.m4
+++ b/config/kernel.m4
@@ -257,9 +257,10 @@ AC_DEFUN([ZFS_AC_SPL], [
 		], [
 			AC_MSG_RESULT([Not found])
 			AC_MSG_ERROR([
-	*** Please make sure the spl devel package for your distribution
-	*** is installed then try again.  If that fails you can specify the
-	*** location of the spl source with the '--with-spl=PATH' option.])
+	*** ZFS depends on the SPL kernel module, which is not found.
+	*** DKMS should retry zfs-dkms momentarily after building spl-dkms.
+	*** If this error message persists, or if you are running DKMS manually,
+	*** then run 'apt-get install --reinstall spl-dkms' and try again.])
 		])
 	], [
 		AS_IF([test "$splsrc" = "NONE"], [
-- 
1.7.9.1

